src/apis/impl/DataSampleApiImpl.ts

```diff
@@ -463,95 +391,93 @@ export class DataSampleApiImpl implements DataSampleApi {
         ...existingService,
         content: updatedContent,
       })!
-    );
+    ).catch(e => {
+      throw this.errorHandler.createErrorFromAny(e)
+    });
 
     return documentId;
   }
 
   async deleteDataSample(dataSampleId: string): Promise<string> {
-    let deletedDataSampleId = (
+    const deletedDataSampleId = (
       await this.deleteDataSamples([dataSampleId])
     ).pop();
     if (deletedDataSampleId) {
       return deletedDataSampleId;
     }
 
-    throw Error(`Could not delete data sample ${dataSampleId}`);
+    throw this.errorHandler.createErrorWithMessage(`Could not delete data sample ${dataSampleId}`)
```

Comment:
Indeed, an error will be thrown at XHR level. Since we call deleteDataSamples, it'll be catch there: https://github.com/icure/icure-medical-device-js-sdk/blob/37f082b52d6f544ddbe793cfc1bb45c44c941314/src/apis/impl/DataSampleApiImpl.ts#L413-L415